### DISCUSSION TITLE
Argument for change to PGN 65379

### DISCUSSION THREAD

**Discussion by cas206** [canboat/canboat#434] [General]
From collected data: 

```
// Command (0 - 204)
2023-12-21T18:30:55.187Z,3,126208,0,204,17,
01, // Command
   63,ff,00,// PGN 0x00ff63 = 65379
   f8, // Priority 8
       // Reserved (f)
   04, // Number of Parameters 4
   01,3b,07,
   03,04,
   04,40,00,
   05,ff,ff

// Group Command Acknowledge
2023-12-21T18:30:55.187Z,3,126208,204,0,8,
   02, // Acknowledge
   63,ff,00, // PGN 0x00ff63 = 65379
   00, // Error Code
       // Transmission Interval
   04, // Number of Parameters
   00, // No Error 1
       // No Error 2
   00  // No Error 3
       // No Error 4

```
The PGN 126208 Command to PGN 65379 is treating the two Pilot Mode (8 bit) / Pilot Submode (8 bit) as a single 16 bit field.  This is followed by another 16 bit field.  Could the Pilot Mode be 16 bits and the Pilot Submode be 16 bits?  Or are they combined into a single 16 bit value, followed by 16 bit data?

The next PGN 65379 Data is: 

```
// PGN 65379
2023-12-21T18:30:55.187Z,7,65379,204,255,8,
   3b,9f, // Field 1,2,3
   40,00, // Field 4 (as seen in Command message)
   00,00, // Field 5 (as seen in Command message)
   02,    // Field 6 ?
   ff     // Field 7 ? / Reserved
```

**Comment by sbender9** [comment]
same comment here. please include the full messages also without comments. 

I'd prefer not to have to copy these and edit them to test. Could be error prone.

**Comment by cas206** [comment]
The data capture file, ydnu-02-raymarine-ev1.dle, is now in samples folder. A file for analyzer can be generated with actisense-serial -r filename. The following is a snippet of the data I used to form my observations in post 1.

`2023-12-23T14:46:47.244Z,3,126208,0,204,17,01,63,ff,00,f8,04,01,3b,07,03,04,04,00,00,05,ff,ff
2023-12-23T14:46:47.244Z,3,126208,204,0,8,02,63,ff,00,00,04,00,00
2023-12-23T14:46:47.244Z,7,126720,204,255,13,3b,9f,f0,81,84,f6,e6,9b,40,00,0e,02,08
2023-12-23T14:46:47.244Z,7,65379,204,255,8,3b,9f,00,00,00,00,02,ff
`

**Comment by TwoCanPlugIn** [comment]
Observations and working code from various OpenCPN plugins indicate that both the Pilot Mode and Pilot Sub Mode fields in PGN 65379 are 16bits.

And the analysis of PGN 126208 above (which is used to command the autopilot), confirms this.

**Reply by cas206** [reply]
So you agree that field 4 Pilot Mode and field 5 Sub Mode for this PGN at https://canboat.github.io/canboat/canboat.html should be changed to "16 bits BINARY"?   So that leaves the last 2 bytes as undetermined (at least by me from the study of my short data sample).

**Reply by TwoCanPlugIn** [reply]
From the samples I have, the second last byte is always 0x02, which clearly has some significance or meaning.

 And the last byte is always 0xFF (meaning unused/data unavailable, and probably just pads the message to 8 bytes)
